### PR TITLE
Shield belts

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Implements shield belts, a researchable wearable item that gives the user an energy shield.
- The shield has 80 total health and will not take damage from disabler beams or other 0 damage weapons. This will absorb about 4 shots from a laser gun and 2 shots from a shotgun (giving melee users some additional time to close the distance between themselves and a target, and a little bit more confidence when it comes to engaging a projectile target).
- The shield does not protect against melee attacks
- The shield is coloured based on its health
- If anyone fires a gun while wearing the shield, they will shoot (and damage) themselves
- The shield is unlocked with the advanced weaponry research node, which along with standard weaponry has been reduced to 5000 research points.
- The shield belt is manufactored in the science protolathe.

Counters:
- EMPs will disable the shield (as if it took 80 damage in 1 go)
- Melee weapons completely bypass the shield, forcing you into melee combat with the wearer

Note that it also blocks thrown items due to the nature of projectile blocking on shields.

## Why It's Good For The Game

Adds a new item that gives both some utility to the crew, and heavy benefits towards mid-game antagonists that don't have projectile weapons. Adds a more interesting way to buff melee against the overwhelmingly powerful projectile based meta, while considering how to not make them too powerful while in the hands of crew/security.

I chose the science protolathe for manufactoring because security is far too though on who they give out their gear, and its more science than security. I chose not to also include it in the sec protolathe so that security explicitly have to ask for scientists to print it by going down there.

I chose to make it use the belt slot (and only the belt slot) since other than the backpack it is the only high-value storage slot where the player stores important items. You have to choose between the extra storage of a belt, an extra slot for a weapon or the shield belt.

I chose to make it completely block 0-damage weapons, since those are typically held by members of the crew, and I want this to be good for the crew but more powerful in the hands of anti-crew. I also dislike the use of stuns when someone is prepared to engage in real life or death combat.

## Testing Photographs and Procedure

image

Shield effect, includes a pulsing animation shown in the video below

https://github.com/user-attachments/assets/b10a0143-1e13-490c-95be-4540924fb0e6

Note that I encountered a bug with shield belt recharging instantly after EMP'd. This has been corrected to take 20 seconds post-EMP to start recharging again.

## Changelog
:cl:
add: Adds the shield belt, a researchable anti-projectile shield that blocks outgoing and incoming projectiles.
balance: Weaponry and advanced weaponry research nodes now cost 5000 instead of 10000.
/:cl:

### CODE REVIEW

**Comment by DarnTheMarn** [comment]
> The shield has 80 total health and will not take damage from disabler beams or other 0 damage weapons. This will absorb about 4 shots from a laser gun and 2 shots from a shotgun 

This makes it sound like it still blocks disabler beams, it just doesn't reduce the shield health. Is that right?

**Comment by PowerfulBacon** [comment]
Yes, like all shields disablers and 0-damage projectiles will not damage the shield

File: code/modules/holoparasite/holoparasite_manifest.dm [hunk 1]
```diff
@@ -32,7 +32,7 @@
 			mr_hud?.begin_timer(HOLOPARASITE_MANIFEST_COOLDOWN)
 		playsound(loc, 'sound/creatures/holopara_summon.ogg', vol = 45, extrarange = HOLOPARA_MANIFEST_SOUND_EXTRARANGE, frequency = 1)
 		add_filter("holopara_manifest", 1, gauss_blur_filter(size = 4))
-		transition_filter("holopara_manifest", 1.2 SECONDS, list("size" = 0), easing = SINE_EASING, loop = FALSE)
+		transition_filter("holopara_manifest", 1.2 SECONDS, list("size" = 0), easing = SINE_EASING, loop = 0)
```
Comment (Rukofamicom) [inline code/modules/holoparasite/holoparasite_manifest.dm:35]:
```suggestion
		transition_filter("holopara_manifest", 1.2 SECONDS, list("size" = 0), easing = SINE_EASING, loop = FALSE)
```

Why was this changed from `FALSE` to `0`?

**Comment by Rukofamicom** [comment]
Seems like a really cool item (or subtype of) for Ninjas to spawn with, given the discussion about their issues with security ranged weapons. I think having something like this would be even better for them overall than the armor adjustments, since it lets them focus less on trying to break into medbay for supplies.

File: code/modules/holoparasite/holoparasite_manifest.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/holoparasite/holoparasite_manifest.dm:35]: Loops is the number of times you want it to loop and not a boolean. It's pedantic, but i thought I'd make it the correct type anyway

Comment (Rukofamicom) [reply code/modules/holoparasite/holoparasite_manifest.dm:35]: Makes sense to me

**Review by Rukofamicom — APPROVED** [review]

**Comment by DarnTheMarn** [comment]
I do wonder how people will choose to abuse it. Security could roll with the classic teleshield + secway + baton + new shield belt, antagonists could run with with it sixteen other ways themselves. Basically a boss battle with three health bars.

**Comment by Rukofamicom** [comment]
The health value of the shield can always be adjusted later if it ends up being too strong

**Review by PowerfulBacon — COMMENTED** [review]
Stamina projectiles damage the shield, this is a bug with cult shields too

**Comment by Rukofamicom** [comment]
Interesting, I wonder why

**Review by Tsar-Salat — CHANGES_REQUESTED** [review]
I think this needs more work.

~~The idea is solid, but theres no audio feedback when your attack is blocked by a shield (consuming a charge or otherwise). There should be a sound/visual cue when projectiles are deflected.~~ EDIT: It just wasnt working for self-inflicted shots it seems?

Additionally, the visual effect surrounding the player doesn't look like a typical shield and more like the effect from holiness/ratvar cultists. The color change I do not think is adequate to show what level its at due the subtlety of it.

Pairing these together, I think the Player Experience is kinda diminished.

I would suggest adding both audio for the various interactions with the shield, as well as maybe the shield "flaring up" across and having a more pronounced visual effect when hit, rather than a continuous subtle one.

**Comment by PowerfulBacon** [comment]
I have to disagree with a few of the points on player experience being diminished:
- Since they are just normal shields, getting shot produces sparks the same way any other shield produces sparks on shot
- The shield colour change is subtle, but you can kind of tell when something happens
- Since the shield is transparent, thicker and pulses it is pretty easy to distinguish as a shield in my opionion.
- When the shield activates and charges, it plays a sound like other shields

I didn't seem to notice any obvious UX issues when the testmerge was applied, though it would need more time to uncover anything.

**Comment by Tsar-Salat** [comment]
> * The shield colour change is subtle, but you can kind of tell when something happens
> * Since the shield is transparent, thicker and pulses it is pretty easy to distinguish as a shield in my opionion.

can you show what the shield looks like when worn by someone with a holy aura or a clock cultist?

**Comment by PowerfulBacon** [comment]
image

image

Since the shield is a filter effect, it applies around any other effect currently applied to your mob, including height displacement, overlays/underlays and mutable appearances.

**Comment by ColonelOrion** [comment]
Lisan al Gaib!
